admin/src/js/components/matches/match.js

```diff
@@ -0,0 +1,134 @@
+import React from 'react';
+import Person from './person';
+
+export default class Match extends React.Component {
+    constructor() {
+        super()
+
+        this.removePerson = this.removePerson.bind(this);
+        this.nagHost = this.nagHost.bind(this);
+    }
+
+    cancelMatch(e) {
+        if (e) {
+            e.preventDefault();
+        }
+
+        return fetch(`/api/match/${this.props.match.id}`, {
+            method: 'DELETE',
+            credentials: 'include',
+            headers: {
+                'Content-Type': 'application/json'
+            }
+        }).then(matches => this.props.fetchMatches());
+    }
+
+    removeBoth(e) {
+        if (e) {
+            e.preventDefault();
+        }
+
+        return fetch(`/api/match/${this.props.match.id}`, {
+            method: 'DELETE',
+            credentials: 'include',
+            headers: {
+                'Content-Type': 'application/json'
+            }
+        }).then(() => {
+            return fetch(`/api/person/${this.props.match.host_id}`, {
+                method: 'DELETE',
+                credentials: 'include',
+                headers: {
+                    'Content-Type': 'application/json'
+                }
+            })
+        }).then(() => {
+            return fetch(`/api/person/${this.props.match.guest_id}`, {
+                method: 'DELETE',
+                credentials: 'include',
+                headers: {
+                    'Content-Type': 'application/json'
+                }
+            })
+        }).then(matches => this.props.fetchMatches());
```

Comment:
Removal of guest and host can be done in parallel with `const removeHost = fetch(...); const removeGuest = fetch(...); Promise.all([removeHost, removeGuest])`